### ISSUE TITLE
Enhance GET _node/stats/pipelines API for Metricbeat monitoring

### ISSUE THREAD

**Issue by ycombinator** [elastic/logstash#10120]
### Background

Today, when the `xpack.monitoring.*` settings are configured in `logstash.yml`, Logstash performs _internal_ collection of it's own monitoring metrics and ships them to `.monitoring-logstash-*`. The Monitoring UI in Kibana then reads documents from these indices to create visualizations of a user's Logstash instances and pipelines.

In the future, we're aiming to have Metricbeat _externally_ collect Logstash metrics and ship them to `.monitoring-logstash-*` indices. This will mean that Metricbeat has to poll various Logstash HTTP APIs to collect said metrics.

### The Request

Currently the `GET _node/stats/pipelines` API returns much of the information that Metricbeat will need in order to build the documents to be indexed into `.monitoring-logstash-*`. However, there are a few additional pieces of information that this API — for each pipeline object under the `pipelines` array in the response — would need to return to complete the picture:

- [x] `ephemeral_id`: Ephemeral ID of pipeline
- [x] `hash`: Hash of pipeline representation
- [x] `queue`: Pipeline queue type + metrics
- [x] `reloads`: Pipeline reload metrics
- [ ] `vertices`: LIR vertices of the pipeline + their metrics

Here is an example JSON document showing **just these additional fields**:

```json
{
  "ephemeral_id":"6d69db08-f2a5-411f-81cf-0f395d5de358",
  "hash":"4a5913a37e548b5921a3a7aec761cc0e59f0ce988c44bca8bcf14163a0cfc550",
  "queue":{
    "type":"memory",
    "queue_size_in_bytes":0,
    "max_queue_size_in_bytes":0,
    "events_count":0
  },
  "reloads":{
    "successes":0,
    "failures":0
  },
  "vertices":[
    {
      "pipeline_ephemeral_id":"6d69db08-f2a5-411f-81cf-0f395d5de358",
      "queue_push_duration_in_millis":0,
      "events_out":0,
      "id":"a6288a968df1bdcfe6433498f8bbabd15e6be53134f3594128330fd37b9ca837"
    },
    {
      "duration_in_millis":16,
      "pipeline_ephemeral_id":"6d69db08-f2a5-411f-81cf-0f395d5de358",
      "events_in":0,
      "events_out":0,
      "id":"991a78728a853623523a25457cf2932267f11f34b0c8af6310c48e44a6d69c43"
    },
    {
      "duration_in_millis":21,
      "pipeline_ephemeral_id":"6d69db08-f2a5-411f-81cf-0f395d5de358",
      "events_in":0,
      "events_out":0,
      "id":"42811067c43d0beafc02fa83221107895efc12eb37f9c4d92b94758bdf1dd3e8"
    }
  ]
}
```

**Comment by jsvd** [comment]
> queue: Pipeline queue type + metrics
> reloads: Pipeline reload metrics

These are already provided:

```
% curl -s localhost:9600/_node/stats/pipelines/main |jq ".pipelines.main.queue"
{
  "type": "memory"
}
% curl -s localhost:9600/_node/stats/pipelines/main |jq ".pipelines.main.reloads"
{
  "last_error": null,
  "successes": 0,
  "last_success_timestamp": null,
  "last_failure_timestamp": null,
  "failures": 0
}
```

For the `vertices` I'd also guard it behind a `?vertices=true` parameter

**Comment by ycombinator** [comment]
Thanks @jsvd. So, just so I'm understanding this correctly, Metricbeat should first call `GET _node/stats/pipelines` to get all the pipeline IDs, then loop over them and call `GET _node/stats/pipelines/:id` for each? I don't suppose you could return all the information in the `GET _node/stats/pipelines` response itself, with `vertices` guarded by `?vertices=true`?

**Comment by ycombinator** [comment]
@cachedout Per the requirements in the issue description, the `GET _node/stats/pipelines` API response needs to return the `ephemeral_id` and `hash` properties under each pipeline object under the `pipelines` array. I see that these items are checked off in the issue description, indicating that at some point this was indeed the case.

However, now that #10576 has been merged, I tried running Logstash from `master` and calling `GET _node/stats/pipelines`. Those two fields are missing. So I'm reopening this issue.

### Steps to reproduce

1. Run Logstash from `master`.
   ```
   ./bin/logstash -e 'input { stdin {} } output { elasticsearch {} }'
   ```
   
2. Visit http://localhost:9600/_node/stats/pipelines in your browser.

3. Verify that `pipelines.main.ephemeral_id` and `pipelines.main.hash` are populated with correct values.

**Comment by cachedout** [comment]
Closed by #10885
